src/daemon/input_linux.c

```diff
@@ -159,32 +167,35 @@ void os_keypress(usbdevice* kb, int scancode, int down){
         event.value = down;
         is_mouse = !!(scancode & SCAN_MOUSE);
     }
-    if(write((is_mouse ? kb->uinput_mouse : kb->uinput_kb) - 1, &event, sizeof(event)) <= 0)
-        ckb_warn("uinput write failed: %s\n", strerror(errno));
-    else
-        isync(kb);
+    if(is_mouse){
```

Comment:
This can be simplified by having a single write call and picking the appropriate handle based on the if.
For example

```C
int fd = (is_mouse ? kb->uinput_mouse : kb->uinput_kb) - 1;
// [...]
if(write(fd, [...]
```

Using the same thing you can also use a unified isync (as mentioned above) and just call isync unconditionally in the end with the fd passed to it.